sdcm/utils/issues.py

```diff
@@ -0,0 +1,87 @@
+
+import github
+import github.Auth
+from github.GithubException import UnknownObjectException
+
+from sdcm.keystore import KeyStore
+from sdcm.sct_config import SCTConfiguration
+from sdcm.sct_events.base import Severity
+from sdcm.sct_events.system import TestFrameworkEvent
+
+scylla_issue_pattern = re.compile(
+    r"^\s*((((?P<user_id>[\w:-]+)/)?(?P<repo_id>[\w:-]+))?#)?(?P<issue_id>\d+)\s*$", re.IGNORECASE)
+
+scylla_issue_https_pattern = re.compile(
+    r"^https?:/.*?/(?P<user_id>[\w:-]+)/(?P<repo_id>[\w:-]+)/.*?/(?P<issue_id>\d+?)$", re.IGNORECASE)
+
+
+@dataclass
+class Issue:
+    user_id: str | None
+    repo_id: str | None
+    issue_id: int | None
+
+
+class SkipPerIssues:
+    def __init__(self, issues: list[str] | str, params: SCTConfiguration | dict):
+        self.params = params
+
+        github_access = KeyStore().get_json("github_access.json")
+        auth = github.Auth.Token(token=github_access["token"])
+        self.github = github.Github(auth=auth)
+
+        issues = [issues] if isinstance(issues, str) else issues
+
+        self.issues = [self.get_issue_details(issue) for issue in issues]
+        self.issues = list(filter(lambda x:  x, self.issues))  # filter None - unmatched issues
+
+    @staticmethod
+    def parse_issue(issue_str):
+        for pattern in [scylla_issue_pattern, scylla_issue_https_pattern]:
+            match = pattern.search(issue_str.strip())
+            if match:
+                obj = match.groupdict()
+                return Issue(
+                    user_id=obj.get("user_id") or "scylladb",
+                    repo_id=obj.get("repo_id") or "scylladb",
+                    issue_id=int(obj.get("issue_id"))
+                )
+        return None
+
+    @lru_cache
+    def get_issue_details(self, issue):
+        issue_parsed = self.parse_issue(issue)
+        if not issue_parsed:
+            # TODO raise event about it
+            logging.warning("couldn't parse issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"couldn't parse issue: {issue}",
+                               severity=Severity.WARNING,
+                               trace=sys._getframe().f_back).publish()  # pylint: disable=protected-access
+            return None
+        try:
+            return self.github.get_user(issue_parsed.user_id).get_repo(issue_parsed.repo_id).get_issue(issue_parsed.issue_id)
+        except UnknownObjectException:
+            logging.warning("couldn't find issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"couldn't find issue: {issue}",
+                               severity=Severity.WARNING,
+                               trace=sys._getframe().f_back).publish()  # pylint: disable=protected-access
+            return None
+
+    def issues_opened(self):
+        return any(issue.state != 'closed' for issue in self.issues)
```

Comment:
when testing Scylla for release branches and issue is not backported won't this allow to Nemesis run and fail?

Reply:
That what the labels are for, to control on which branch we disable enable, it would be ontop of the issue, and not depended on SCT code being backported or not.

Most of those are Scylla issues, and enabled/disabling something in SCT won't be dependent on backport in SCT